core/lib/config/src/configs/external_price_api_client.rs

```diff
@@ -0,0 +1,29 @@
+use std::time::Duration;
+
+use serde::Deserialize;
+
+pub const DEFAULT_TIMEOUT_MS: u64 = 10_000;
+
+#[derive(Debug, Clone, PartialEq, Deserialize)]
+pub struct ExternalPriceApiClientConfig {
+    #[serde(default = "ExternalPriceApiClientConfig::default_source")]
+    pub source: String,
+    pub base_url: Option<String>,
```

Comment:
Once again, there should be no defaults for these values.
Configs are loaded conditionally, e.g. they are loaded as an `Option` by default, and only dereferenced if actually used. We only use this config if we run base token ratio persister -- a new component -- so we don't have to worry about backward compatibility.

At the same time, declaring these fields as optional can lead to unwanted and unexpected results. For example, we want to run a "real" price fetcher, but we forgot to provide configs. Default configs will be loaded, and the behavior will _likely_ not match what we want.

There is no default that makes sense "in general"; for localhost it's no-op impl, for production it's coingecko, they are both equally important.

So I suggest that you:
a) make `source` required.
b) remove `default_source()` method.
c) add localhost defaults to the dev configs (both env and file-based).

This shouldn't be breaking because, as I said, this is a new component, and the config won't be accessed unless it's needed.